airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -86,7 +90,46 @@ def upload_file_if_changed(
     return False, remote_blob.id
 
 
-def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path) -> Tuple[bool, str]:
+def _latest_upload(metadata: ConnectorMetadataDefinitionV0, bucket: storage.bucket.Bucket, metadata_file_path: Path) -> Tuple[bool, str]:
+    version_path = get_metadata_remote_file_path(metadata.data.dockerRepository, metadata.data.dockerImageTag)
+    latest_path = get_metadata_remote_file_path(metadata.data.dockerRepository, "latest")
+    latest_icon_path = get_icon_remote_file_path(metadata.data.dockerRepository, "latest")
+
+    (
+        version_uploaded,
+        version_blob_id,
+    ) = upload_file_if_changed(metadata_file_path, bucket, version_path)
+    latest_uploaded, _latest_blob_id = upload_file_if_changed(metadata_file_path, bucket, latest_path)
+
+    # Replace metadata file name with icon file name
+    local_icon_path = metadata_file_path.parent / ICON_FILE_NAME
+    if local_icon_path.exists():
+        upload_file_if_changed(local_icon_path, bucket, latest_icon_path)
+
+    return version_uploaded or latest_uploaded, version_blob_id
+
+
+def _prerelease_upload(metadata: ConnectorMetadataDefinitionV0, bucket: storage.bucket.Bucket, prerelease_tag: str) -> Tuple[bool, str]:
+    # replace any dockerImageTag references with the actual tag
+    # this includes metadata.data.dockerImageTag, metadata.data.registries[].dockerImageTag
+    # where registries is a dictionary of registry name to registry object
+    metadata_dict = to_json_sanitized_dict(metadata, exclude_none=True)
+    metadata_dict["data"]["dockerImageTag"] = prerelease_tag
+    for registry in get(metadata_dict, "data.registries", {}).values():
+        if "dockerImageTag" in registry:
+            registry["dockerImageTag"] = prerelease_tag
+
+    # write metadata to yaml file in system tmp folder
+    tmp_metadata_file_path = Path("/tmp") / metadata.data.dockerRepository / prerelease_tag / METADATA_FILE_NAME
+    tmp_metadata_file_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(tmp_metadata_file_path, "w") as f:
+        yaml.dump(metadata_dict, f)
+
+    prerelease_remote_path = get_metadata_remote_file_path(metadata.data.dockerRepository, prerelease_tag)
+    return upload_file_if_changed(tmp_metadata_file_path, bucket, prerelease_remote_path, disable_cache=True)
```

Comment:
why disable_cache, and only on prereleases?

Reply:
I had disabled the cache here since I figured metadata might change more rapidly. But youre right, we should disable it all the time